language/diem-framework/modules/0L/Teams.move

```diff
@@ -0,0 +1,302 @@
+///////////////////////////////////////////////////////////////////////////
+// 0L Module
+// Teams 
+///////////////////////////////////////////////////////////////////////////
+// Used for coordinating "Teams", equivalent to delegation
+// Teams are groups which engage in work.
+// some of that work is automated such as validation.
+// A Team has a collective "consensus weight", which is used for voting in consensus
+// but also for stdlib upgrades.
+// Team members can share in a validator's rewards, but can also be used for coordinating
+// other activity, like projects and bounties.
+// File Prefix for errors: TBD
+///////////////////////////////////////////////////////////////////////////
+
+address 0x1 {
+module Teams {
+    use 0x1::CoreAddresses;
+    use 0x1::TowerState;
+    use 0x1::Vector;
+    use 0x1::Signer;
+    use 0x1::DiemAccount;
+    use 0x1::ValidatorUniverse;
+    use 0x1::Decimal;
+    use 0x1::Testnet;
+    use 0x1::Errors;
+
+    const ENOT_SLOW_WALLET: u64 = 1010;
+    
+    struct AllTeams has key, copy, drop, store {
+      teams_list: vector<address>, // the team is identified by its captain.
+      collective_threshold_epoch: u64,
+      member_threshold_epoch: u64,
+      tower_height_rms: u64,
+    }
+
+    struct Team has key, copy, drop, store {
+      captain: address, // A validator account. TODO this is redundant, since it is stored in the validator account. But placed here for future-proofing.
+      members: vector<address>,
+      operator_pct_reward: u64, // the percentage of the rewards that the captain proposes to go to the validator operator.
+      collective_tower_height_this_epoch: u64,
+
+      // Informational fields
+      team_name: vector<u8>, // A validator account.
+      description: vector<u8>,
+      count_all_members: u64, // count the miners who have expressed to me members of this team, but may be below threshold
+      count_active: u64, // the count of players above threshold
+
+      // Is team in validator set
+    }
+
+    // this struct is stored in the member's account
+    struct Member has key, copy, drop, store {
+      captain_address: address, // by address of captain
+      mining_above_threshold: bool, // if the mining the user has done is above the system threshold to count toward delegation.
+
+    }
+
+    public fun vm_init(sender: &signer) {
+      CoreAddresses::assert_vm(sender);
+      if (!exists<AllTeams>(CoreAddresses::VM_RESERVED_ADDRESS())) {
+        move_to<AllTeams>(
+          sender, 
+          AllTeams {
+            teams_list: Vector::empty(),
+            collective_threshold_epoch: 0,
+            member_threshold_epoch: 0,
+            tower_height_rms: 0,
+          }
+        );
+      }
+    }
+
+
+    public fun team_init(sender: &signer, team_name: vector<u8>, operator_pct_reward: u64) {
+
```

Comment:
Need to confirm that the operator_pct_reward is between 0 and 100

Reply:
fixed with: 970536449cf3bc1e2d0b236ffb53fff4297a25c3